### PR TITLE
`Assessment` : Add the possibility for instructors to drop an assessment instruction for new feedback

### PR DESCRIPTION
**Pull request by Olha-D**
<!-- Thanks for contributing to Artemis! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist
#### General
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable and you may remove the first checkmark. If you are unsure, please test on the test servers. -->
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] Following the [theming guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
Solves the Issue #5252 

### Description
<!-- Describe your changes in detail -->
I have added the possibility to drop an assessment instruction in the area under the submission (Add new Feedback button and beneath). On this event a function is called, that automatically creates a new feedback and adds the information of the dropped instruction to it.  
### Steps for Testing
<!-- Please describe in detail how the reviewer can test your changes. -->
Prerequisites:
- 1 Instructor
- 1 Programming, Quiz, Modelling or File upload exercise with Assessment instructions 
- 1 submitted solution of the exercise

1. Log in to Artemis
2. Navigate to Course Management
3. Find the course with the modelling exercise and click on the Submissions button near the  exercise. 
4. Click Open Assessment to the right from one of the submissions.
5. Now you should be able to drag one of the Assessment instructions to the area under the submission and drop it there. Then a new Feedback should appear in the list of Feedbacks with the information from the Assessment instruction. 

### Review Progress
<!-- Each Pull Request should be reviewed by at least two other developers. The code as well as the functionality (= manual test) needs to be reviewed. -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
<!-- Please add the test coverages for all changed files here. You can see this when executing the tests locally (see build.gradle and package.json) or when looking into the corresponding Bamboo build plan. -->
<!-- Lines are the main reference but a significantly lower branch percentage can indicate missing edge cases in the tests. -->
| Class/File | Branch | Line |
|------------|-------:|-----:|
|unreferenced-feedback.component.ts | 83% |  96% |                                                                                                                                                                                                                                                                                                                                                     



### Screenshots
<!-- Add screenshots to demonstrate the changes in the UI. -->
<!-- Create a GIF file from a screen recording in a docker container https://toub.es/2017/09/11/high-quality-gif-with-ffmpeg-and-docker/ -->
![DropInstruction](https://user-images.githubusercontent.com/69432438/179007077-d690d001-14e1-416b-b233-bd21942af0f5.gif)

### CODE REVIEW

**Comment by Strohgelaender** [comment]
Could you also allow drag and drop for the whole section below that where all the feedback elements are listed? I think dragging something on a button is not quite obvious.

**Comment by ge65cer** [comment]
I agree with Dominik, we should keep the features consistent across all applicable exercise types.

File: src/main/webapp/app/exercises/shared/unreferenced-feedback/unreferenced-feedback.component.ts [hunk 1]
```diff
@@ -27,6 +28,8 @@ export class UnreferencedFeedbackComponent {
 
     @Output() feedbacksChange = new EventEmitter<Feedback[]>();
 
+    constructor(public structuredGradingCriterionService: StructuredGradingCriterionService) {}
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/unreferenced-feedback/unreferenced-feedback.component.ts:31]:
```suggestion
    constructor(private structuredGradingCriterionService: StructuredGradingCriterionService) {}
```

File: src/main/webapp/app/exercises/shared/unreferenced-feedback/unreferenced-feedback.component.ts [hunk 2]
```diff
@@ -90,4 +93,13 @@ export class UnreferencedFeedbackComponent {
         });
         return Math.max(...references.concat([0])) + 1;
     }
+
+    createAssessmentOnDrop(event: Event) {
+        this.addUnreferencedFeedback();
+        const newFeedback: Feedback | undefined = this.unreferencedFeedback.last();
+        if (newFeedback !== undefined) {
```
Comment (Strohgelaender) [inline src/main/webapp/app/exercises/shared/unreferenced-feedback/unreferenced-feedback.component.ts:100]:
```suggestion
        if (newFeedback) {
```

**Review by dfuchss — APPROVED** [review]
Tested in Testing Session. Works as expected.

**Review by Dominik-Weinzierl — DISMISSED** [review]
Looks good to me :+1:

**Review by sleiss — APPROVED** [review]
Tested in the testing session.

**Review by dfuchss — APPROVED** [review]
Re-Approve

**Review by nairaarnez — APPROVED** [review]
Tested in ts1, it works as described!

**Review by ge65cer — CHANGES_REQUESTED** [review]
See my comments below.

**Review by jschmautzer — DISMISSED** [review]
tested on ts1, worked as described

**Review by babyygemperor — APPROVED** [review]
Tested on ts1, works as described

**Review by nairaarnez — DISMISSED** [review]
Tested in TS1

**Review by jschmautzer — DISMISSED** [review]
re-approve, tested on ts2

**Review by ge65cer — DISMISSED** [review]
Approve code.

File: src/main/webapp/i18n/de/fileUploadAssessment.json [hunk 3]
```diff
@@ -6,6 +6,7 @@
             "notFound": "Wir haben keine neue Datei-Upload-Abgabe ohne Bewertung gefunden, bitte gehe zurück.",
             "addFeedback": "Neues Feedback hinzufügen",
             "assessInstruction": "Bitte klicke auf \"Neues Feedback hinzufügen\".",
+            "assessInstructionWithDropOption": "Bitte klicke auf \"Neues Feedback hinzufügen\" oder ziehen Sie eine Bewertungsanweisung und legen Sie sie hier ab.",
```
Comment (Strohgelaender) [inline src/main/webapp/i18n/de/fileUploadAssessment.json:9]:
```suggestion
            "assessInstructionWithDropOption": "Bitte klicke auf \"Neues Feedback hinzufügen\" oder ziehe eine Bewertungsanweisung und lege sie hier ab.",
```

**Review by JohannesStoehr — APPROVED** [review]
Code and screencast look good

**Review by nairaarnez — APPROVED** [review]
Re-tested in ts1. Re-approved

**Review by jschmautzer — APPROVED** [review]
re-approve, tested on ts1

**Review by MetinLamby — APPROVED** [review]
tested on ts1, works as shown in the video

**Review by luisa-we — APPROVED** [review]
Tested on TS1, works as expected!

**Pull request merged**